### PR TITLE
fix: new overflow items should only be enqueued while observing

### PR DESCRIPTION
**Pull request by ling1726**
## Current Behavior

Currently overflow items are enqueued immediately when added to the manager. This can create subtle bugs because all options are set during `.observe()`. Some of those options can alter the priority of the items in the queue.

## New Behavior

Adds a new `observing` flag to the overflow manager and enqueues all overflow items during `observe()` any new items added while `observing` is true will be enqueued directly.

Also adds tests

## Related Issue(s)

Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                        |    Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                 Change |
| :----------------------------------------------------------------------------------------------------------------------- | --------------------------: | --------------------------: | -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>priority-overflow</samp> <br /> <abbr title='bundle-size/overflow-manager.fixture.js'>createOverflowManager</abbr> |  `2.936 kB`<br />`1.212 kB` |  `2.996 kB`<br />`1.237 kB` | `60 B` <br />`25 B`  |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                          | `10.685 kB`<br />`4.104 kB` | `10.745 kB`<br />`4.124 kB` | `60 B` <br />`20 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `188.815 kB`<br />`52.287 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |    `33.4 kB`<br />`11.008 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/64fb6212cd2f12d40f10f16d0154b5786c3534c6'>64fb6212cd2f12d40f10f16d0154b5786c3534c6</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/Avatar-mount_base.html">1286</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/Avatar-mount.html">1259</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/Button-mount_base.html">926</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/Button-mount.html">917</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/FluentProvider-mount_base.html">1517</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/FluentProvider-mount.html">1489</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/FluentProviderWithTheme-mount_base.html">572</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/FluentProviderWithTheme-mount.html">566</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">534</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">530</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">574</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">579</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/MakeStyles-mount_base.html">1939</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/MakeStyles-mount.html">1958</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/SpinButton-mount_base.html">2316</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25122/perf-test-react-components/SpinButton-mount.html">2334</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-priority-overflow-faf497c0-15d0-4c21-b8cf-b5d7fdeafd9f.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "new overflow items should only be enqueued while observing",
```
Comment (layershifter) [inline change/@fluentui-priority-overflow-faf497c0-15d0-4c21-b8cf-b5d7fdeafd9f.json:3]:
```suggestion
  "comment": "fix: new overflow items should only be enqueued while observing",
```

File: packages/react-components/priority-overflow/src/overflowManager.ts [hunk 2]
```diff
@@ -162,6 +163,9 @@ export function createOverflowManager(): OverflowManager {
 
   const observe: OverflowManager['observe'] = (observedContainer, userOptions) => {
     Object.assign(options, userOptions);
+    observing = true;
```
Comment (layershifter) [inline packages/react-components/priority-overflow/src/overflowManager.ts:166]: Should be this set to `false` in `disconnect`?

**Review by layershifter — APPROVED** [review]

File: packages/react-components/priority-overflow/src/overflowManager.ts (hunk 2 above)
Comment (ling1726) [reply packages/react-components/priority-overflow/src/overflowManager.ts:166]: yeah good catch

**Pull request merged**